src/coreclr/ilasm/CMakeLists.txt

```diff
@@ -107,6 +117,16 @@ else()
     )
 endif(CLR_CMAKE_TARGET_WIN32)
 
+if(CLR_CMAKE_TARGET_UNIX AND NOT CLR_CMAKE_TARGET_APPLE)
+  list(APPEND ILASM_LINK_LIBRARIES System.Security.Cryptography.Native.OpenSsl-Static)
```

Comment:
Are you suggesting I use `System.Security.Cryptography.Native.OpenSsl` instead? I'm probably going about this in a naive way, but setting `GEN_SHARED_LIB` to true and trying to link this library in doesn't work for me locally.

Reply:
I think what you have in the PR is correct.

`System.Security.Cryptography.Native.OpenSsl-Static` is statically linked version of the shim, but it still loads the opensll dynamically in distro agnostic build config.

Reply:
So long as FEATURE_DISTRO_AGNOSTIC_SSL is set that does appear to be true, based on

https://github.com/dotnet/runtime/blob/1478ff6913c48af8566ea8bb8f1239b7a2b78a6b/src/native/libs/System.Security.Cryptography.Native/extra_libs.cmake#L34C9-L39

Since this file has an if in it around that "feature flag", I don't know which way we're building.